app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt

```diff
@@ -12,45 +12,119 @@ import androidx.compose.foundation.clickable
+ * a snackbar will be displayed.
  *
  * @param wallpapers Wallpapers to add to grid.
  * @param selectedWallpaper The currently selected wallpaper.
- * @param onSelectWallpaper Action to take when a new wallpaper is selected.
+ * @param onSelectWallpaper Callback for when a new wallpaper is selected.
+ * @param onViewWallpaper Callback for when the view action is clicked from snackbar.
  */
 @Composable
 fun WallpaperSettings(
     wallpapers: List<Wallpaper>,
     selectedWallpaper: Wallpaper,
     onSelectWallpaper: (Wallpaper) -> Unit,
+    onViewWallpaper: () -> Unit,
 ) {
-    Surface(color = FirefoxTheme.colors.layer2) {
-        WallpaperThumbnails(
-            wallpapers = wallpapers,
-            selectedWallpaper = selectedWallpaper,
-            onSelectWallpaper = onSelectWallpaper
-        )
+    val coroutineScope = rememberCoroutineScope()
+    val scaffoldState = rememberScaffoldState()
+
+    Scaffold(
+        backgroundColor = FirefoxTheme.colors.layer1,
+        scaffoldState = scaffoldState,
+        snackbarHost = { hostState ->
+            SnackbarHost(hostState = hostState) {
+                WallpaperSnackbar(onViewWallpaper)
+            }
+        }
+    ) {
+        Surface(color = FirefoxTheme.colors.layer2) {
+            WallpaperThumbnails(
+                wallpapers = wallpapers,
+                selectedWallpaper = selectedWallpaper,
+                onSelectWallpaper = { updatedWallpaper ->
+                    coroutineScope.launch {
+                        scaffoldState.snackbarHostState.showSnackbar(
+                            message = "", // overwritten by WallpaperSnackbar
+                            duration = SnackbarDuration.Short
+                        )
+                    }
+                    onSelectWallpaper(updatedWallpaper)
+                },
+            )
+        }
     }
 }
 
+@Composable
+private fun WallpaperSnackbar(
+    onViewWallpaper: () -> Unit,
+) {
+    Snackbar(
+        modifier = Modifier
+            .padding(8.dp)
+            .heightIn(min = 48.dp),
+        backgroundColor = FirefoxTheme.colors.actionPrimary,
+        content = {
+            Text(
+                modifier = Modifier.padding(start = 16.dp, end = 16.dp, top = 8.dp, bottom = 8.dp),
+                text = stringResource(id = R.string.wallpaper_updated_snackbar_message),
+                textAlign = TextAlign.Start,
+                color = FirefoxTheme.colors.textInverted,
+                fontFamily = FontFamily(Font(R.font.metropolis_semibold)),
+                fontSize = 18.sp,
+            )
+        },
+        action = {
+            TextButton(
+                modifier = Modifier.padding(start = 16.dp, end = 16.dp, top = 8.dp, bottom = 8.dp),
+                onClick = onViewWallpaper,
+            ) {
+                Text(
+                    text = stringResource(id = R.string.wallpaper_updated_snackbar_action),
```

Comment:
Let's drop the named parameter `id`. I know I like to see named parameters, but given that it is a bit obvious what goes into stringResource, and how we didn't do this for the other stringResource (let's be consistent). Named parameters do help when the code isn't obvious without the help an IDE such as `doSomething(true, false, true, 1, 0)`.